pkg/cmd/server/bootstrappolicy/all_test.go

```diff
@@ -0,0 +1,121 @@
+package bootstrappolicy
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+const osClusterRoleAggregationPrefix = "system:openshift:"
+
+// this map must be manually kept up to date as we make changes to aggregation
+// we hard code this data with no constants because we cannot change the underlying values
+var expectedAggregationMap = map[string]sets.String{
+	"admin": sets.NewString("system:openshift:aggregate-to-admin", "system:aggregate-to-admin"),
+	"edit":  sets.NewString("system:openshift:aggregate-to-edit", "system:aggregate-to-edit"),
+	"view":  sets.NewString("system:openshift:aggregate-to-view", "system:aggregate-to-view"),
+}
+
+func TestPolicyAggregation(t *testing.T) {
+	policyData := Policy()
+	clusterRoles := policyData.ClusterRoles
+	clusterRolesToAggregate := policyData.ClusterRolesToAggregate
+
+	// do some basic sanity checks
+
+	if len(clusterRoles) == 0 || len(clusterRolesToAggregate) == 0 {
+		t.Fatalf("invalid policy data:\n%#v\n%#v", clusterRoles, clusterRolesToAggregate)
+	}
+
+	// make sure we have no duplicate new names
+	shouldHaveAggregationRuleSet := sets.NewString()
+	newNameOfClusterRoleSet := sets.NewString()
+	for oldName, newName := range clusterRolesToAggregate {
+		if newNameOfClusterRoleSet.Has(newName) {
+			t.Errorf("duplicate value %s for key %s", newName, oldName)
+		}
+		shouldHaveAggregationRuleSet.Insert(oldName)
+		newNameOfClusterRoleSet.Insert(newName)
+	}
+
+	// now the actual test
+
+	hasAggregationRuleSet := sets.NewString()
+	aggregationMap := map[string]sets.String{} // map of cluster role name to all cluster roles that aggregate into it
+	for i := range clusterRoles {
+		cr := clusterRoles[i]
+		if cr.AggregationRule == nil {
+			continue
+		}
+
+		hasAggregationRuleSet.Insert(cr.Name)
+
+		// insert this cluster role into aggregationMap with all of cluster roles that aggregate into it
+		for j := range cr.AggregationRule.ClusterRoleSelectors {
+			labelSelector := cr.AggregationRule.ClusterRoleSelectors[j]
+			selector, err := v1.LabelSelectorAsSelector(&labelSelector)
+			if err != nil {
+				// should never happen
+				t.Errorf("invalid label selector %#v   at index %d for cluster role %s: %v", labelSelector, j, cr.Name, err)
```

Comment:
are these extra spaces in the message (`%#v   at index`) were added intentionally?